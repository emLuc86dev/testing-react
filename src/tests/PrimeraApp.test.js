import React from "react";
import "@testing-library/jest-dom";
import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Testing PrimeraApp", () => {
  test("should return <PrimeraApp /> ", () => {
    const greeting = `Hi, I'm Emilio`;

    const wrapper = shallow(<PrimeraApp saludo={greeting} />);

    // expect(wrapper).toMatchSnapshot()
    expect(wrapper).toMatchSnapshot();
  });

  test("should show subtitle sent by props", () => {
    const subtitle = "Subtitle sent by props";
    const greeting = `Hi, I'm Emilio`;
    const wrapper = shallow(
      <PrimeraApp saludo={greeting} subtitulo={subtitle} />
    );

    // expect(toJson(wrapper)).toMatchSnapshot();

    const textP = wrapper.find("p").text();
    expect(textP).toEqual(subtitle);

  });
});
