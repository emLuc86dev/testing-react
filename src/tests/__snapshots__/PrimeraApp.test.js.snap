// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PrimeraApp should return <PrimeraApp />  1`] = `
<Fragment>
  <h1>
     
    Hi, I'm Emilio
     !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
